base/libgit2/callbacks.jl

```diff
@@ -78,68 +99,40 @@ function authenticate_ssh(libgit2credptr::Ptr{Ptr{Void}}, p::CredentialPayload,
+            isnull(response) && return user_abort()
+            privatekey = unsafe_get(response)
+
+            # Only update the public key if the private key changed
+            if privatekey != creds.prvkey
+                publickey = privatekey * ".pub"
             end
-            keydefpath
         end
 
-        isfile(privatekey) || warn("Private key not found")
-
-        # If the private key changed, invalidate the cached public key
-        (privatekey != creds.prvkey) &&
-            (creds.pubkey = "")
-
-        # For SSH we need a public key location, look for environment vars SSH_* as well
-        publickey = if haskey(ENV,"SSH_PUB_KEY_PATH")
-            ENV["SSH_PUB_KEY_PATH"]
-        else
-            keydefpath = creds.pubkey # check if credentials were already used
-            if isempty(keydefpath)
-                if isempty(keydefpath)
-                    keydefpath = privatekey*".pub"
-                end
-                if !isfile(keydefpath)
-                    response = Base.prompt("Public key location for '$prompt_url'",
-                        default=keydefpath)
-                    isnull(response) && return user_abort()
-                    keydefpath = unsafe_get(response)
-                end
-            end
-            keydefpath
+        # For SSH we need a public key location. Avoid asking about the public key as
+        # typically this will just annoy users.
+        if !isfile(publickey) && isfile(privatekey)
+            response = Base.prompt("Public key location for '$prompt_url'",
+                default=publickey)
+            isnull(response) && return user_abort()
+            publickey = unsafe_get(response)
         end
 
-        passphrase = if haskey(ENV,"SSH_KEY_PASS")
-            ENV["SSH_KEY_PASS"]
-        else
-            passdef = creds.pass # check if credentials were already used
-            if isempty(passdef) && is_passphrase_required(privatekey)
-                if Sys.iswindows()
-                    response = Base.winprompt(
-                        "Your SSH Key requires a password, please enter it now:",
-                        "Passphrase required", privatekey; prompt_username = false)
-                    isnull(response) && return user_abort()
-                    passdef = unsafe_get(response)[2]
-                else
-                    response = Base.prompt("Passphrase for $privatekey", password=true)
-                    isnull(response) && return user_abort()
-                    passdef = unsafe_get(response)
-                    isempty(passdef) && return user_abort()  # Ambiguous if EOF or newline
-                end
+        if isempty(passphrase) && is_passphrase_required(privatekey)
+            if Sys.iswindows()
+                response = Base.winprompt(
+                    "Your SSH Key requires a password, please enter it now:",
+                    "Passphrase required", privatekey; prompt_username = false)
```

Comment:
Spacing.

Reply:
Issue here is `prompt_username = false`?

Reply:
Yeah, and a few others. Looking at other places in base I'm guessing it should be.

```julia
response = Base.winprompt("Your SSH Key requires a password, ",
                                              "please enter it now: Passphrase required",
                                              privatekey; prompt_username=false)
```

Reply:
One of the more extreme cases of this is in `authenticate_userpass`: 
```julia
response = Base.winprompt("Please enter your credentials for '$prompt_url'", "Credentials required",
    isempty(username) ? p.username : username; prompt_username = true)
```
which would become the following in order to fit within the 92-limit:
```julia
response = Base.winprompt("Please enter your credentials for '$prompt_url'",
                          "Credentials required", 
                          isempty(username) ? p.username : username; 
                          prompt_username=true)
```
I think this is also acceptable in Base (I've been allowed to do this before): 
```julia
response = Base.winprompt(
    "Please enter your credentials for '$prompt_url'", "Credentials required", 
    isempty(username) ? p.username : username; prompt_username=true)
```
I'll note that the last example just slightly exceeds the 92-limit but I think is probably the most readable.

I'll try to strike a balance.

Reply:
Thanks.